src/Imghoard/ImghoardClient.cs

```diff
@@ -58,34 +55,30 @@ public async Task<IReadOnlyList<Image>> GetImagesAsync(params string[] Tags)
                 }
             }
 
-            Uri url;
 
-            if (query != null)
-                url = new Uri(API_Base, $"/images?tags={query}");
-            else
-                url = new Uri(API_Base, $"/images");
-
-            var response = await APIClient.GetAsync(url.ToString());
+            StringBuilder urlBuilder = new StringBuilder("/images");
+            if(query != null)
+            {
+                urlBuilder.Append($"?tags={query}");
+            }
+            var response = await apiClient.GetAsync(urlBuilder.ToString());
 
             if (response.Success)
             {
                 return JsonConvert.DeserializeObject<IReadOnlyList<Image>>(response.Body);
             }
 
+            // TODO(velddev): Add better error handling.
             throw new Exception(response.HttpResponseMessage.ReasonPhrase);
         }
         
         public async Task<Image> GetImageAsync(ulong Id)
         {
-            var url = new Uri(API_Base, $"/images?id={Id}");
-
-            var response = await APIClient.GetAsync(url.ToString());
-
+            var response = await apiClient.GetAsync("/images?id={Id}");
```

Comment:
The string interpolation (`$`) has been removed, thus the output is always `/images?id={Id}`.